sessiontxn/isolation/base.go

```diff
@@ -247,13 +270,16 @@ func (p *baseTxnContextProvider) ActivateTxn() (kv.Transaction, error) {
 	return txn, nil
 }
 
-func (p *baseTxnContextProvider) prepareTxn() error {
+func (p *baseTxnContextProvider) prepareTxn(considerSnapshotTS bool) error {
```

Comment:
I think we can seperate prepareTxn(bool) to `prepareTxn` and `prepareTxnWithOracleTS`